src/code/views/quick-action-menu.tsx

```diff
@@ -0,0 +1,202 @@
+                onClick={this.toggleShowImages}
+                onMouseEnter={showImagesF}
+              >
+                <div>{setImageLabel}</div>
+                <div className="icon-codap-inspectorArrow-collapse"/>
+              </li>
+
+              { isTimeBased &&
+                <>
+                  <li
+                    className={collectorClasses}
+                    onClick={this.convertToCollector}
+                    onMouseEnter={hideImagesF}
+                  >
+                    {collectorLabel}
+                  </li>
+
+                  <li
+                    className={flowClasses}
+                    onClick={this.convertToFlow}
+                    onMouseEnter={hideImagesF}
+                  >
+                    {flowLabel}
+                  </li>
+
+                  <li
+                    className={standardClasses}
+                    onClick={this.convertToStandard}
+                    onMouseEnter={hideImagesF}
+                  >
+                  {standardVariableLabel}
+                  </li>
+                </>
+              }
+
+              {showGraphButton &&
+                <li
+                  className={createGraphClasses}
+                  onClick={this.createGraph}
+                  onMouseEnter={hideImagesF}
+                >
+                  {createGraphLabel}
+                </li>
+              }
+
+            </ul>
+          </div>
+          {true && this.renderRightMenu()}
+        </div>
+      </>
+    );
+  }
+
+  private renderRightMenu() {
+    const thisNode = this.props.node;
+    const selected = thisNode.image;
+    const onImageChage = (i: {image: string}) =>  {
+      GraphStore.changeNode({ image: i.image }, thisNode);
+    };
+
+    const palette = PaletteStore.orderedPalette();
+    const imageAddCallback = (i: {image: string}) => {
+      this.setState({ showImages: true, closeTimer: null });
+      GraphStore.changeNode({ image: i.image }, thisNode);
+    };
+    return (
+      <div className="right-panel">
+        <div className="image-choices">
+          <PaletteAddView
+            label={tr("~PALETTE-INSPECTOR.ADD_IMAGE")}
+            callback={imageAddCallback}
+          />
+          { palette.map((pi, i) =>
+            <div className="image-choice-wrapper" key={i}>
+              <ImgChoiceView
+                key={i}
+                node={pi}
+                selected={selected}
+                onChange={onImageChage}
+              />
+            </div>
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  private handleMouseEnter = () => {
+    if (this.state.closeTimer) {
+      clearTimeout(this.state.closeTimer);
+    }
+    this.setState({ closeTimer: null });
+  }
+
+  private handleMouseLeave = () => {
+    const {closeFn} = this.props;
+    if (closeFn) {
+      const nextTimeout = setTimeout(() => {
+        closeFn();
+      }, 500);
+      this.setState({ closeTimer: nextTimeout });
```

Comment:
Do you want to keep `closeTimer` in the component state?  This will cause a re-render.  No big deal, but just wondering.